source/nvwave.py

```diff
@@ -341,22 +453,29 @@ def close(self):
 			with self._waveout_lock:
 				if not self._waveout:
 					return
-				# We're explicitly closing, so kill the close timer.
-				if self._closeTimer and self._closeTimer.IsRunning():
-					self._callOnMainThread(self._closeTimer.Stop)
 				self._close()
 
 	def _close(self):
+		log.debug("Calling winmm.waveOutClose")
 		with self._global_waveout_lock:
-			winmm.waveOutClose(self._waveout)
+			if not self._waveout:
+				return
+			try:
+				winmm.waveOutClose(self._waveout)
+			except WindowsError:
+				log.debug("Error closing the device, it may have been removed.", exc_info=True)
 		self._waveout = None
 
 	def __del__(self):
 		self.close()
 		winKernel.kernel32.CloseHandle(self._waveout_event)
 		self._waveout_event = None
 

```

Comment:
Although not technically to do with this pr, but might you please be able to add the following call to super. It is necessary for garbageHandler.TrackedObject to work at all. I missed this previously.
```Suggestion
		super().__del__()

```